FirebaseAuth/Sources/Auth/FIRAuth.m

```diff
@@ -457,12 +460,13 @@ + (FIRAuth *)authWithApp:(FIRApp *)app {
   return (FIRAuth *)auth;
 }
 
-- (instancetype)initWithApp:(FIRApp *)app {
+- (instancetype)initWithApp:(FIRApp *)app appCheck:(nullable id<FIRAppCheckInterop>)appCheck {
```

Comment:
How about passing the appcheck directly to requesetConfiguration instead of modifying Auth initializer?

Reply:
Wouldn't it still have to go through the initializer? Since AppCheck is injected into FirebaseAuth via FirebaseComponents.

Reply:
Then I suggest pass it via FIRApp (similar to heartbeatLogger), so we can keep the initializer as simple as possible.

Reply:
Thanks for the advice, But there will be more changes in other files like FIRApp and FIRAppInternal, if trying to pass it in FIRApp. I don't think we should add any logic into FIRApp.